app/controllers/api/messaging.php

```diff
@@ -0,0 +1,2145 @@
+
+        $message = $dbForProject->createDocument('messages', new Document([
+            '$id' => $messageId,
+            'to' => $to,
+            'description' => $description,
+            'data' => [
+                'content' => $content,
+            ],
+            'status' => $status,
+            'search' => $messageId . ' ' . $description,
+        ]));
+
+        if ($status === 'processing') {
+            $queueForMessaging
+                ->setMessageId($message->getId())
+                ->setProject($project)
+                ->trigger();
+        }
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->dynamic($message, Response::MODEL_MESSAGE);
+    });
+
+App::post('/v1/messaging/messages/push')
+    ->desc('Create a push notification.')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'message.create')
+    ->label('audits.resource', 'message/{response.$id}')
+    ->label('scope', 'messages.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'createPushNotification')
+    ->label('sdk.description', '/docs/references/messaging/create-push-notification.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MESSAGE)
+    ->param('messageId', '', new CustomId(), 'Message ID. Choose a custom ID or generate a random ID with `ID.unique()`. Valid chars are a-z, A-Z, 0-9, period, hyphen, and underscore. Can\'t start with a special char. Max length is 36 chars.')
+    ->param('to', [], new ArrayList(new Text(Database::LENGTH_KEY), 1), 'List of Topic IDs or List of User IDs or List of Target IDs.')
+    ->param('title', '', new Text(256), 'Title for push notification.')
+    ->param('body', '', new Text(64230), 'Body for push notification.')
+    ->param('description', '', new Text(256), 'Description for Message.', true)
+    ->param('data', null, new JSON(), 'Additional Data for push notification.', true)
+    ->param('action', '', new Text(256), 'Action for push notification.', true)
+    ->param('icon', '', new Text(256), 'Icon for push notification. Available only for Android and Web Platform.', true)
+    ->param('sound', '', new Text(256), 'Sound for push notification. Available only for Android and IOS Platform.', true)
+    ->param('color', '', new Text(256), 'Color for push notification. Available only for Android Platform.', true)
+    ->param('tag', '', new Text(256), 'Tag for push notification. Available only for Android Platform.', true)
+    ->param('badge', '', new Text(256), 'Badge for push notification. Available only for IOS Platform.', true)
+    ->param('status', 'processing', new WhiteList(['draft', 'processing']), 'Message Status. Value must be either draft or processing.', true)
```

Comment:
Reminder. Let's ensure this works well with SDK enums.

Reply:
These will work